### PR TITLE
Visualise subqueries

### PR DESCRIPTION
**Pull request by lawben**
Subqueries are now visualised in the LQP. PQP will come when I'm further down the line with correlated subqueries. But that will be a separate PR then.

Viz now:
iterm2 pqgloj lqp

The Projection has a "right" child, which is the subquery.

### CODE REVIEW

**Comment by mrks** [comment]
Should we make a visual distinction? Maybe a dashed line with "scalar subquery" on it?

**Comment by Bouncner** [comment]
Yes, I would definitely miss ghe fact that this Q includes a subquery.

**Comment by lawben** [comment]
Updated image in description.

File: src/lib/planviz/lqp_visualizer.cpp [hunk 1]
```diff
@@ -82,11 +98,16 @@ void LQPVisualizer::_build_dataflow(const std::shared_ptr<AbstractLQPNode>& from
     label_stream << "no est.";
   }
 
-  VizEdgeInfo info = _default_edge;
-  info.label = label_stream.str();
+  VizEdgeInfo info = edge_info;
+  info.label = edge_info.label.empty() ? label_stream.str() : edge_info.label;
```
Comment (mrzzzrm) [inline src/lib/planviz/lqp_visualizer.cpp]: This is a hacky (and therefore hard to read) way to introduce label overrides. You're calculating all this (somewhat nonsense) info that goes into label_stream, only to discard it here.

File: src/lib/planviz/lqp_visualizer.cpp [hunk 2]
```diff
@@ -46,47 +48,70 @@ void LQPVisualizer::_build_subtree(const std::shared_ptr<AbstractLQPNode>& node,
...
+
+  if (const auto projection = std::dynamic_pointer_cast<ProjectionNode>(node)) {
+    for (const auto& column_expression : projection->column_expressions()) {
+      if (column_expression->is_subselect()) {
+        _build_subtree(column_expression->subselect_node(), visualized_nodes);
+
+        auto edge_info = _default_edge;
+        edge_info.label = "Scalar Subquery";
+        edge_info.style = "dashed";
+        _build_dataflow(column_expression->subselect_node(), node, edge_info);
```
Comment (mrzzzrm) [inline src/lib/planviz/lqp_visualizer.cpp]: Maybe I am missing something but why don't you just call _add_edge here directly?
The current way _build_dataflow is implemented is still ugly, imo

Comment (lawben) [reply src/lib/planviz/lqp_visualizer.cpp]: Yea, good point. I should have seen this straight away.

File: src/lib/planviz/abstract_visualizer.hpp [hunk 3]
```diff
@@ -39,6 +39,7 @@ struct VizEdgeInfo {
   std::string color = "white";
   std::string font_color = "white";
   double pen_width = 1.0;
+  std::string style;
```
Comment (mrzzzrm) [inline src/lib/planviz/abstract_visualizer.hpp]: This not having a default makes it default to solid line? Maybe add the default? 

After this is solved, mergytime ;)

Comment (lawben) [reply src/lib/planviz/abstract_visualizer.hpp]: problem is: I have no idea what the default string is ;)

**Pull request merged**